### PR TITLE
Bandaid fixes Cleanbot and Kittens

### PR DESCRIPTION
**Pull request by KazeEspada**
Implements a system for living mobs to handle themselves if they are spawned in nullspace.

Had to go to work before even compile testing this.

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -7,9 +7,24 @@ Sorry Giacom. Please don't be mad :(
...
+#DEFINE ADMINWARNNUM	     100 //How many mobs in nullspace before we warn the admins that the server is about to reboot
+#DEFINE CREWWARNNUM 	 	 300 //How many mobs in nullspace before we warn the players that server is about to reboot
+#DEFINE SERVERRESTARTNUM	 500 //Reboot the server to prevent the imminent crash
+
+/mob/living
+	var/static/mobsInNullSpace = 0
 
 /mob/living/New()
 	. = ..()
+	if(isnull(get_turf(loc))
```
Comment (PKPenguin321) [inline code/modules/mob/living/living.dm]: You're short a )

**Comment by KazeEspada** [comment]
Theres still more compile issues, but like I said I am at work so don't wanna fix them right now.

**Comment by lzimann** [comment]
is it a good idea have an automatic reboot like that that admins cant even stop?

**Comment by KazeEspada** [comment]
Server is on the road to crashing anyways if you have 500 mobs spawning in nullspace.

**Comment by lzimann** [comment]
i've seen way more than 500 without crashing

**Comment by KazeEspada** [comment]
Give me some numbers?

**Comment by lzimann** [comment]
it's been a while since i've seen it happening, but the first time there was like 8k cleanbots in nullspace

**Comment by duncathan** [comment]
only keep the admin warning. the reboot is pointless but notifying the admins will be useful for debugging

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -7,9 +7,24 @@ Sorry Giacom. Please don't be mad :(
...
+
+/mob/living
+	var/static/mobsInNullSpace = 0
 
 /mob/living/New()
 	. = ..()
+	if(isnull(get_turf(loc)))
+		mobsInNullSpace++
+		if(mobsInNullSpace == ADMINWARNNUM)
+			warn_admins("Alert! There are currently over [ADMINWARNNUM] mobs that spawned in nullspace. An automatic server reboot may be imminent."
```
Comment (PKPenguin321) [inline code/modules/mob/living/living.dm]: missing another )

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -7,9 +7,31 @@ Sorry Giacom. Please don't be mad :(
...
+		if(mobsInNullSpace == DEFCON1)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 1.")
+		if(mobsInNullSpace == DEFCON2)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 2.")
+		if(mobsInNullSpace == DEFCON3)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 3.")
+		if(mobsInNullSpace == DEFCON4)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 4.")
+		if(mobsInNullSpace == DEFCON5)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 5. It is strongly recommended that the server before forced to restart.")
```
Comment (Iamgoofball) [inline code/modules/mob/living/living.dm]: I'm gonna say this should probably be made a world message in userdanger spans so everyone knows "welp shit is fucked

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -7,9 +7,31 @@ Sorry Giacom. Please don't be mad :(
...
 		push_mob_back(src, A.push_dir)
 */
+#define DEFCON1		200
+#define DEFCON2  	400
+#define DEFCON3		600
+#define DEFCON4		800
+#define DEFCON5		1000
+
+/mob/living
+	var/static/mobsInNullSpace = 0
```
Comment (Iamgoofball) [inline code/modules/mob/living/living.dm]: why is this var on every living mob

Comment (KazeEspada) [inline code/modules/mob/living/living.dm]: Its a static var, so all mobs share it. Its on every living mob cause i dont know the extent of the cleanbot.

Comment (lordpidey) [inline code/modules/mob/living/living.dm]: shouldn't it be global?  Static seems inappropriate for a value that changes once set.

File: code/modules/mob/living/living.dm (hunk 3 above)
Comment (lordpidey) [inline code/modules/mob/living/living.dm]: IT'S OVER ONE THOUSAAAAANDDDD!!!!!

File: code/modules/mob/living/living.dm (hunk 4 above)
Comment (duncathan) [inline code/modules/mob/living/living.dm]: static is a standard keyword in several languages for a variable shared between all instances of the class. either word is fine; they're synonymous to byond anyway

File: code/modules/mob/living/living.dm [hunk 5]
```diff
@@ -7,9 +7,31 @@ Sorry Giacom. Please don't be mad :(
...
+		if(mobsInNullSpace == DEFCON2)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 2.")
+		if(mobsInNullSpace == DEFCON3)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 3.")
+		if(mobsInNullSpace == DEFCON4)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 4.")
+		if(mobsInNullSpace == DEFCON5)
+			message_admins("Alert! There are currently over [mobsInNullSpace] mobs that spawned in nullspace. This is DEFCON 5. It is strongly recommended that the server before forced to restart.")
+		stack_trace("Mob being instantiated in nullspace.")
+		del(src) //There are 1000 kittens in nullspace, let's not clog up the GC with them.
```
Comment (duncathan) [inline code/modules/mob/living/living.dm]: `qdel(src)`
and then return

**Comment by KorPhaeron** [comment]
@KazeEspada 

Let me know if you want to finish this/have it reopened

**Pull request closed without merging**